src/components/form/signup.js

```diff
@@ -61,83 +92,205 @@ const LoginLink = styled.a`
 class Signup extends Component {
     state = {
         email: '',
+        is_submitting: false,
+        error_msg: '',
+        submit_status: '',
     }
 
     handleInputChange = e => {
-        e.preventDefault()
         const { name, value } = e.target
 
         this.setState({
             [name]: value,
         })
+        this.validateEmail()
```

Comment:
Can we pass the email from state to the `validateEmail` function? since this.setState is asynchronous and email validation is one step behind